addons/account/models/account_account.py

```diff
@@ -97,11 +112,12 @@ def _check_account_type_unique_current_year_earning(self):
         help="Check this box if this account allows invoices & payments matching of journal items.")
     tax_ids = fields.Many2many('account.tax', 'account_account_tax_default_rel',
         'account_id', 'tax_id', string='Default Taxes',
-        check_company=True,
+        domain="[('company_id', 'parent_of', company_ids)]",
         context={'append_type_to_tax_name': True})
     note = fields.Text('Internal Notes', tracking=True)
-    company_id = fields.Many2one('res.company', string='Company', required=True, readonly=False,
+    company_ids = fields.Many2many('res.company', string='Companies', required=True, readonly=False,
```

Comment:
Does the ORM prevent against creating an account without companies even if it is required here? (AFAIK it only works in client)

Reply:
No, but I added a constraint to prevent this.